src/crypto.ts

```diff
@@ -87,7 +91,15 @@ export default class Crypto extends CryptoBase {
       }
 
       if (verifiedContent) {
-        if (!this.signingPublicKey) {
+        if (!this.getSigningPublicKey) {
```

Comment:
Can this be done on the constructor instead? The crypto class should not be instantiated if there's no function provided. This allows us to throw the error earlier rather than later.

Reply:
I was doing exactly this in the earlier commits! but then it broke this [test](https://github.com/opengovsg/formsg-javascript-sdk/blob/develop/spec/crypto.spec.ts#L216-L241), where it allows class instantiation without a public key but "disallows" certain function being called (....by throwing an error).

I assume it's a tradeoff made at the time? So to keep behaviour backward compatible I decided not too mess around with that area of the code.

But I feel this is somewhat a confusing pattern, as every potential downstream method of the class have to be wary of that property. 

Anyways since we're jumping to a major version already (from 0.13.0 -> 1.0), we can consider changing this particular behaviour and adjusting that test :p

Reply:
Agreed. I feel that it doesn't really make sense to allow incomplete instantiation resulting in downstream function having to cleanup.

Reply:
_paiseh just got time to dedicate wrapping up work on this PR_.

@KenLSM just had a revelation why it's coded this way - SDK is used in 3 places:
1. Form frontend: admin page, for decrypting responses. Doesn't need public signing key.
2. Form backend: for crafting signatures andallthat. Doesn't need public signing key.
3. Webhook consumer: for verifying said signatures. Need public signing key.

So the same class is conditionally instantiated for 2 different usecases, which in nature requires different prerequisite.

Let me think the best way to approach this, if the pursuit of cleanliness ends up introducing a fat refactor on the main form codebase then I would defer the change and take baby steps instead.